### DISCUSSION TITLE
Error "Invalid option for ..." in HA core logs

### DISCUSSION THREAD

**Discussion by stomko11** [Koenkk/zigbee2mqtt#11043] [Devices]
I am not sure if this is issue with the device, Zigbee2MQTT or Home Assistant, so I am submitting that as a question. I am seeing following errors in my HA core logs:
```
2022-01-26 20:32:15 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.office_e6_sensor_temperature_unit_convert: '°C' (valid options: ['celsius', 'fahrenheit'])
```
Basically that repeats for all of my sensors. I am using the dev branch of Zigbee2MQTT, as this introduced the support for Nous E6 sensors.
If I go to Z2MQTT - select the device - Exposes - and select "celsius" there, the display on the devices switches to Fahrenheit

<img width="388" alt="image" src="https://user-images.githubusercontent.com/88718231/151234057-7a596c50-4ccd-4e98-b8d2-76e8cc966129.png">

And I also get additional log:
```
2022-01-26 20:37:12 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.office_e6_sensor_temperature_unit_convert: '' (valid options: ['celsius', 'fahrenheit'])
2022-01-26 20:37:12 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'temperature_unit_convert' when rendering '{{ value_json.temperature_unit_convert }}'
```

Other than that, device works fine, I get proper temperature readings, etc., it is just that my core logs are flooded with these errors and I would like to get rid of them.

**Comment by roumano** [comment]
I confirm the issue : 
```
oct. 05 00:08:10 pi hass[666683]: 2023-10-05 00:08:10.589 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'temperature_unit' when rendering '{{ value_json.temperature_unit }}'
oct. 05 00:08:10 pi hass[666683]: 2023-10-05 00:08:10.591 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.soil_sensor_temperature_unit: '' (valid options: ['celsius', 'fahrenheit'])
```

Found how to reproduce it easily : 
- add, as normal, a device on zigbee2mqtt
- after, few dozon of secondes, (the time it's creating mqtt topics and time for Home Assistant to discovery it)
- Go into Setting of the sensors, add a field into "Filtered publish attributes" , then submit
  - in my exemple it's `temperature_unit` :
```
# grep -B3 temperature_unit /data/zigbee2mqtt/data/configuration.yaml 
  '0xa4c1380243ac3161':
    friendly_name: Soil sensor
    filtered_attributes:
      - temperature_unit
```

**Reply by Koenkk** [reply]
What device is this `Soil sensor`?

**Reply by roumano** [reply]
Hi,
It's a `TuYa : TS0601_soil`
But i don't think it's a issue with this device specificity
I get the same message on other device where i also added a "Filtered publish attributes".

at least on these device i'm able to reproduce the error : 
- Nue / 3A : HGZB-42-UK / HGZB-41 / HGZB-41-UK
- Nue : 3A - HGZB-41
- Zemismart : LXN56-SS27LX1.1
- LELLKI : WK34-EU
- TuYa: TS011F_plug_3

**Reply by Koenkk** [reply]
it should be fixed after setting the `temperature_unit` once.

**Comment by roumano** [comment]
Yes but my `TuYa: TS011F_plug_3`, i put `outage_memory:`, `child_lock` and `indicator_mode` in the "Filtered publish attributes" ( because i don't want/need to have it in HA as i will never change it, or if i need to change i will use the zigbee2mqtt interface)

But for it, i got the same kind of error in HA : 
````
oct. 05 20:09:43 pi hass[709939]: 2023-10-05 20:09:43.337 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.chargeur_velo_power_outage_memory: '' (valid options: ['on', 'off', 'restore'])

oct. 05 20:09:43 pi hass[709939]: 2023-10-05 20:09:43.339 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'indicator_mode' when rendering '{{ value_json.indicator_mode }}'

oct. 05 20:09:43 pi hass[709939]: 2023-10-05 20:09:43.340 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.chargeur_velo_indicator_mode: '' (valid options: ['off', 'off/on', 'on/off', 'on'])

oct. 05 20:09:43 pi hass[709939]: 2023-10-05 20:09:43.341 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'child_lock' when rendering '{{ value_json.child_lock }}'
```

**Comment by SeLLeRoNe** [comment]
I see the same with a PIR/Presence sensor from Tuya:
`2023-12-28 13:49:01.302 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.first_floor_bathroom_radar_tumble_switch: '1' (valid options: ['ON', 'OFF'])
`

It seems to basically flood the logs with that and considering I have that same device in 5 rooms, the "noise" is high :)

**Reply by Koenkk** [reply]
For what device is this?

**Reply by SeLLeRoNe** [reply]
Apologies, I missed this response!
It's a mmWave radar: [MIR-HE200-TY](https://www.zigbee2mqtt.io/devices/MIR-HE200-TY.html#tuya-mir-he200-ty)

**Reply by SeLLeRoNe** [reply]
Hi @Koenkk , did you have a chance to see this? Yeah, very old thing but still happening :)

**Reply by andrei-lazarov** [reply]
@SeLLeRoNe someone with the device needs to fix it. If you want, open an issue and post full Z2M **debug** logs.
I see there is a mistake in the code, but I don't know what the device transmits

**Reply by SeLLeRoNe** [reply]
I see, I thought it was just about what z2m sends over MQTT, since I have about 200 ecies, is there a way to debut just that device?

Thanks

**Reply by andrei-lazarov** [reply]
No.. only if you have a spare Zigbee dongle. Start a temporary Z2M instance.
But we can usually search by the device name to see the relevant messages.

**Reply by SeLLeRoNe** [reply]
So, I did set the logs to debug, clicked on "OFF" on the Tumble Switch in z2m and the only log line I see is the MQTT output, am I missing something?

<img width="518" height="351" alt="image" src="https://github.com/user-attachments/assets/d012c929-3894-4af0-8c9e-f00f15588de9" />

`[07/04/2026, 16:38:06] z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Studio Andrea Radar', payload '{"device":{"applicationVersion":72,"dateCode":"","friendlyName":"Studio Andrea Radar","hardwareVersion":1,"ieeeAddr":"0xSOMETHING-SOMETHING","manufacturerID":4098,"manufacturerName":"_TZE200_lu01t0zl","model":"MIR-HE200-TY","networkAddress":58349,"powerSource":"Mains (single phase)","stackVersion":0,"type":"Router","zclVersion":3},"elapsed":103868,"fall_down_status":"none","fall_sensitivity":1,"illuminance":13,"last_seen":"2026-04-07T14:38:02.176Z","linkquality":72,"motion_direction":"standing_still","motion_speed":0,"occupancy":false,"presence":false,"radar_scene":"default","radar_sensitivity":3,"static_dwell_alarm":null,"tumble_alarm_time":0,"tumble_switch":1}'
`

<img width="1011" height="349" alt="image" src="https://github.com/user-attachments/assets/7a1dfccc-eed3-49ff-9db7-c4658e913c01" />

The reason I thought this was just MQTT related is because here I can see "tumble_switch":1 in the payload, so I thought z2m "translate" a "1" to ON when it's a binary sensor information.

But anyway, is there something more specific I need to do to see "more" debug output?

**Reply by SeLLeRoNe** [reply]
Just noticed something "odd"

`[07/04/2026, 16:43:13] z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Studio Andrea Radar', payload '{"device":{"applicationVersion":72,"dateCode":"","friendlyName":"Studio Andrea Radar","hardwareVersion":1,"ieeeAddr":"0XSOMETHING-SOMETHING","manufacturerID":4098,"manufacturerName":"_TZE200_lu01t0zl","model":"MIR-HE200-TY","networkAddress":58349,"powerSource":"Mains (single phase)","stackVersion":0,"type":"Router","zclVersion":3},"fall_down_status":"none","fall_sensitivity":1,"illuminance":19,"last_seen":"2026-04-07T14:42:33.435Z","linkquality":153,"motion_direction":"standing_still","motion_speed":0,"occupancy":false,"presence":false,"radar_scene":"default","radar_sensitivity":3,"static_dwell_alarm":null,"tumble_alarm_time":0,"tumble_switch":"OFF"}'
[07/04/2026, 16:43:15] z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Studio Andrea Radar', payload '{"device":{"applicationVersion":72,"dateCode":"","friendlyName":"Studio Andrea Radar","hardwareVersion":1,"ieeeAddr":"SOMETHING","manufacturerID":4098,"manufacturerName":"_TZE200_lu01t0zl","model":"MIR-HE200-TY","networkAddress":58349,"powerSource":"Mains (single phase)","stackVersion":0,"type":"Router","zclVersion":3},"elapsed":37074,"fall_down_status":"none","fall_sensitivity":1,"illuminance":19,"last_seen":"2026-04-07T14:43:10.509Z","linkquality":153,"motion_direction":"standing_still","motion_speed":0,"occupancy":false,"presence":false,"radar_scene":"default","radar_sensitivity":3,"static_dwell_alarm":null,"tumble_alarm_time":0,"tumble_switch":1}'`

In this 2 lines you can see it is sending OFF and then back on, so I guess MAYBE z2m is sending OFF as "outgoing request" and then whatever the device respond (I guess "1"), z2m just forward that on to the MQTT message

**Reply by andrei-lazarov** [reply]
Debug logs are not sent over the network to Z2M user interface (to prevent traffic).
Get them from the host machine HA -> Apps -> Z2M -> Log, or from the log directory inside Z2M installation/data folder.

Please create a new issue where you upload them.

**Comment by rodders99** [comment]
Same problem here,

 ERROR (MainThread) [homeassistant.components.mqtt.select] Invalid option for select.master_socket_backlight_mode: 'LOW' (valid options: ['low', 'medium', 'high'])

**Reply by Koenkk** [reply]
For what device is this?

**Reply by rodders99** [reply]
This one: 

https://www.zigbee2mqtt.io/devices/CMA30036.html#clicksmart%252B-cma30036

Is the case of the option the issue? i.e. 'on' vs 'ON'

**Reply by Koenkk** [reply]
Can you try with latest z2m and change the backlight mode once to see if that fixes the issue?

**Reply by rodders99** [reply]
Hi Koenkk, thanks for replying, and happy new year to you.

You can close this, the 'feature' of the backlight is not something I use. I was tracking down and fixing errors being reported and this was the last one, so I just changed the logging level and it's gone :-)

Thanks
